### PR TITLE
convert VirtualizedList stories helpers file to TS

### LINKED ISSUE #2514: docs: convert VirtualizedList stories helpers file to TS
Convert the `VirtualizedList` stories helpers file to TypeScript:

- Change the file extension to tsx
- Resolve type issues (if there are), avoid using `any` and prefer existing types

## Related files
`packages/core/src/components/VirtualizedList/__stories__/VirtualizedList.stories.helpers.js`

---

[Monday ticket [internal]](https://monday.monday.com/boards/3532714909/views/80492480/pulses/7623433691)

### PR DESCRIPTION
**Pull request by Piyush-Tilokani**
- [x] I have read the [Contribution Guide](../packages/core/CONTRIBUTING.md) for this project.

Resolves #2514 

Converted the `packages/core/src/components/VirtualizedList/__stories__/VirtualizedList.stories.helpers.js` file to .tsx 

Please review and merge
Do let me know if any corrections are required

Thank you

### CODE REVIEW

**Review by rivka-ungar — CHANGES_REQUESTED** [review]
Thanks @Piyush-Tilokani! 
Please use existing type `VirtualizedListItem` instead of creating a new type

**Comment by Piyush-Tilokani** [comment]
Updated in the latest commit

File: packages/core/src/components/VirtualizedList/__stories__/VirtualizedList.stories.helpers.tsx [hunk 1]
```diff
@@ -2,17 +2,19 @@
 import React from "react";
 import { Tip } from "vibe-storybook-components";
 
-export const generateItems = (defaultItemSize = 30, itemsCount, layout) => {
-  const items = [];
+export const generateItems = (defaultItemSize: number = 30, itemsCount: number, layout: string) => {
```
Comment (talkor) [inline packages/core/src/components/VirtualizedList/__stories__/VirtualizedList.stories.helpers.tsx:5]: Please remove the type from `defaultItemSize` as it inferred from the default value
See [CI check](https://github.com/mondaycom/vibe/actions/runs/11410064924/job/31787409356?pr=2531#step:6:152)

**Review by talkor — COMMENTED** [review]
Thank you @Piyush-Tilokani ! Let's just remove the redundant type as CI fails

File: packages/core/src/components/VirtualizedList/__stories__/VirtualizedList.stories.helpers.tsx (hunk 1 above)
Comment (Piyush-Tilokani) [reply packages/core/src/components/VirtualizedList/__stories__/VirtualizedList.stories.helpers.tsx]: Made the change in the latest commit
please check if it is correct

**Review by talkor — APPROVED** [review]
Awesome, thanks @Piyush-Tilokani !

**Pull request merged**